Services/EmployeeSalesService.cs

```diff
@@ -0,0 +1,47 @@
+using System.Data;
+using Microsoft.AspNetCore.Mvc;
+
+public class EmployeeSalesService
+{
+    private DataClient _connection;
+    public EmployeeSalesService(DataClient connection){
+        _connection = connection;
+    }
+    public List<EmployeeSalesModel>? GetsalesByNameAndyear(string name, DateTime year){
+        try{
+            var result = _connection.GetResultsFromQuery<EmployeeSalesModel>(
+                "SELECT soh.SalesOrderID, soh.OrderDate, soh.DueDate, soh.ShipDate, soh.Status, "+
+                "sod.SalesOrderDetailID, soh.CustomerID, sp.SalesYTD, sp.SalesLastYear, " +
+                "pe.FirstName, pe.MiddleName, pe.LastName " +
+                "FROM Sales.SalesOrderHeader soh " +
+                "INNER JOIN Sales.SalesOrderDetail sod ON soh.SalesOrderID = sod.SalesOrderID "+
+                "LEFT JOIN Sales.SalesPerson sp ON soh.SalesPersonID = sp.BusinessEntityID "+
+                "INNER JOIN HumanResources.Employee hr ON sp.BusinessEntityID = hr.BusinessEntityID " +
+                "INNER JOIN Person.Person pe ON hr.BusinessEntityID =pe.BusinessEntityID " +
+                "WHERE " + 
+                $"    ('{name}' ='' OR '{name}' IS NULL OR CONCAT(pe.FirstName, ' ', pe.MiddleName, ' ', pe.LastName) LIKE '%{name}%') " +
```

Comment:
This approach is indeed thoughtful as it addresses null values, but there's a challenge. If null values are passed in the parameters within the URI, the API returns a 500 error. To mitigate this, the approach should be complemented by declaring default values or validating null values for 'name' and 'year' at the beginning of the method.